main/broadcasters.c

```diff
@@ -464,12 +464,81 @@ static broadcaster_ops_t beewi_smart_door_ops = {
     .metadata_get = beewi_smart_door_metadata_get,
 };
 
+/* ATC1441 Firmware for the Xiaomi Thermometer LYWSD03MMC
+    Temperature and Humidity Sensor
+   See: https://github.com/atc1441/ATC_MiThermometer */
+
+#define ATC1441_TEMP_HUM_SERVICE_UUID 0x181A
+
+typedef struct {
+    uint16_t not_used;
+    uint16_t service_uuid;
+    mac_addr_t mac;
+    int16_t temp;
+    uint8_t humid;
+    uint8_t battery_percent;
+    uint16_t battery_mv;
+    uint8_t message_counter;
+} __attribute__((packed)) atc1441_temp_hum_t;
+
+static int atc1441_temp_hum_is_broadcaster(uint8_t *adv_data,
+    size_t adv_data_len)
+{
+    uint8_t len;
+    uint8_t *data = esp_ble_resolve_adv_data(adv_data,
+        ESP_BLE_AD_TYPE_SERVICE_DATA, &len);
+
+    atc1441_temp_hum_t *atc1441_data= (atc1441_temp_hum_t *)adv_data;
+
+    if (len < 15 ||
```

Comment:
Are you sure about this `15`? I counted 17 bytes in the `atc1441_temp_hum_t` structure. Perhaps it would be better to check `len != sizeof(atc1441_data)`?